### PR TITLE
doc: documentation errors in cloud provider access

### PR DESCRIPTION
**Pull request by milh0use**
## Description

First commit - fixes to typos and syntax errors in documentation
Second commit - suggested extra information to help customers translate the well documented Atlas Console method of cloud integrations with the sparsely documented Terraform method.

Link to any related issue(s):

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [x] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

### CODE REVIEW

File: website/docs/r/cloud_provider_access.markdown [hunk 1]
```diff
@@ -77,6 +73,8 @@ resource "mongodbatlas_cloud_provider_access_setup" "test_role" {
 * `last_updated_date`                - Date and time when this Azure Service Principal was last updated. This parameter expresses its value in the ISO 8601 timestamp format in UTC.
 * `role_id`                        - Unique ID of this role.
 
+* -> **NOTE:** the two sub-attributes of the aws_config block will likely be needed on the external AWS account side as the constraints on the trust policy of an IAM role you're granting Atlas access to. They can be accessed via `mongodbatlas_cloud_provider_access_setup.<role>`, for example the trust policy will reference `mongodbatlas_cloud_provider_access_setup.<role>.aws_config[0].atlas_aws_account_arn` in its `Principal` statement, and the `mongodbatlas_cloud_provider_access_setup.<role>.aws_config[0].atlas_assumed_role_external_id` in its `Condition.StringEquals.sts:ExternalId` statement.
```
Comment (jwilliams-mongo) [inline website/docs/r/cloud_provider_access.markdown]: minor suggestions: apply formatting to `aws_config` plus adjust one run-on sentence.
```suggestion
* -> **NOTE:** the two sub-attributes of the `aws_config block` will likely be needed on the external AWS account side as the constraints on the trust policy of an IAM role you're granting Atlas access to. They can be accessed via `mongodbatlas_cloud_provider_access_setup.<role>`. For example, the trust policy will reference `mongodbatlas_cloud_provider_access_setup.<role>.aws_config[0].atlas_aws_account_arn` in its `Principal` statement, and the `mongodbatlas_cloud_provider_access_setup.<role>.aws_config[0].atlas_assumed_role_external_id` in its `Condition.StringEquals.sts:ExternalId` statement.
```

**Review by jwilliams-mongo — APPROVED** [review]
@milh0use LGTM % some tiny tweaks

File: website/docs/r/cloud_provider_access.markdown (hunk 1 above)
Comment (AgustinBettati) [inline website/docs/r/cloud_provider_access.markdown]: We have an AWS example which helps understand the usage of both attributes: https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/examples/atlas-cloud-provider-access/aws/aws-roles.tf#L19. Curious if you had seen this example, and if this note is worth adding having this defined.

**Review by AgustinBettati — COMMENTED** [review]
Just a small consideration, all other changes LGTM. Thanks!

File: website/docs/r/cloud_provider_access.markdown (hunk 1 above)
Comment (milh0use) [reply website/docs/r/cloud_provider_access.markdown]: Ah, I didn't know about those examples. Is it worth putting a link to them in the documentation? Or is the link there already and I didn't see it?

Comment (AgustinBettati) [reply website/docs/r/cloud_provider_access.markdown]: Yes agree the examples are not very accessible from our docs. I believe we could add a link to the example I referenced within the `Example Usage with AWS` section, and remove this Note that was added. cc: @Zuhairahmed

Comment (milh0use) [reply website/docs/r/cloud_provider_access.markdown]: Got it - I've changed the note to a link

File: website/docs/r/cloud_provider_access.markdown [hunk 2]
```diff
@@ -77,6 +73,8 @@ resource "mongodbatlas_cloud_provider_access_setup" "test_role" {
 * `last_updated_date`                - Date and time when this Azure Service Principal was last updated. This parameter expresses its value in the ISO 8601 timestamp format in UTC.
 * `role_id`                        - Unique ID of this role.
 
+-> **NOTE:** Code examples of how the mongodbatlas_cloud_provider_access_setup resource and its attributes should be used to permit Atlas to access resources in your own cloud provider accounts are in Github, for [AWS](https://github.com/mongodb/terraform-provider-mongodbatlas/tree/master/examples/atlas-cloud-provider-access/aws) and for [Azure](https://github.com/mongodb/terraform-provider-mongodbatlas/tree/master/examples/atlas-cloud-provider-access/azure)
```
Comment (AgustinBettati) [inline website/docs/r/cloud_provider_access.markdown]:
```suggestion
-> **NOTE:** For more details on how attributes are used to enable access to cloud provider accounts see [AWS example](https://github.com/mongodb/terraform-provider-mongodbatlas/tree/master/examples/atlas-cloud-provider-access/aws) and [Azure example](https://github.com/mongodb/terraform-provider-mongodbatlas/tree/master/examples/atlas-cloud-provider-access/azure). 
```
Small rephrasing to simplify.

**Review by AgustinBettati — APPROVED** [review]
LGTM

**Review by Zuhairahmed — APPROVED** [review]
This is a great catch @milh0use! LGTM on your PR and have confirmed that you have signed our [contributor's agreement](http://www.mongodb.com/contributor). Thanks again!

**Pull request merged**